daffodil-test/src/test/scala/org/apache/daffodil/section07/variables/TestVariables.scala

```diff
@@ -60,6 +60,17 @@ class TestVariables {
   @Test def test_varInstance_10(): Unit = { runner.runOneTest("varInstance_10") }
   @Test def test_varInstance_11(): Unit = { runner.runOneTest("varInstance_11") }
   @Test def test_varInstance_12(): Unit = { runner.runOneTest("varInstance_12") }
+  @Test def test_varInstance_13(): Unit = { runner.runOneTest("varInstance_13") }
+
+  @Test def test_varDirection_1(): Unit = { runner.runOneTest("varDirection_1") }
+  @Test def test_varDirection_2(): Unit = { runner.runOneTest("varDirection_2") }
+  @Test def test_varDirection_3(): Unit = { runner.runOneTest("varDirection_3") }
+  @Test def test_varDirection_err1(): Unit = { runner.runOneTest("varDirection_err1") }
+  @Test def test_varDirection_err2(): Unit = { runner.runOneTest("varDirection_err2") }
+  @Test def test_varDirection_setVar1(): Unit = { runner.runOneTest("varDirection_setVar1") }
+  @Test def test_varDirection_setVar2(): Unit = { runner.runOneTest("varDirection_setVar2") }
+  @Test def test_varDirection_nvi1(): Unit = { runner.runOneTest("varDirection_nvi1") }
+  @Test def test_varDirection_nvi2(): Unit = { runner.runOneTest("varDirection_nvi2") }
 
```

Comment:
Was playing around with this NVI stuff and testing various variable things. Was testing this language in the spec:
> If the instance is not assigned a new default value then it inherits the default value specified by dfdl:defineVariable or externally provided by the DFDL processor.
I don't think we support the "or externaly provided" part of this. Here's the example schema I used:

```xml
  <xs:annotation>
    <xs:appinfo source="http://www.ogf.org/dfdl/">
      <dfdl:format ref="ex:GeneralFormat" />
      <dfdl:defineVariable name="var1" type="xs:int" defaultValue="1" external="true" />
    </xs:appinfo>
  </xs:annotation>

  <xs:element name="root">
    <xs:complexType>
      <xs:sequence>
        <xs:element name="beforeNVI" type="xs:int" dfdl:inputValueCalc="{ $ex:var1 }" />
        <xs:sequence>
          <xs:annotation>
            <xs:appinfo source="http://www.ogf.org/dfdl/">
              <dfdl:newVariableInstance ref="ex:var1" />
            </xs:appinfo>
          </xs:annotation>
          <xs:element name="afterNVI" type="xs:int" dfdl:inputValueCalc="{ $ex:var1 }" />
        </xs:sequence>
      </xs:sequence>
    </xs:complexType>
  </xs:element>
```
If I run the following:
```
echo -n | daffodil parse -Dvar1=2 -s src/main/resources/com/tresys/csv/xsd/csv.dfdl.xsd
```
Then I get the infoset:
```xml
<root>
  <beforeNVI>2</beforeNVI>
  <afterNVI>1</afterNVI>
</root>
```
So the newVariableInstance is using the defalut value from the defineVariable, but my reading of that line in the spec says the default value for the NVI should be the value that was externally provided, so both beforeNVI and afterNVI should be 2.

Is my reading correct? If so, I don't think that needs to be fixed as part of this, but we should open a bug.

Reply:
Definitely a bug. Externally set variables that have a default value, the default value is 100% discarded and the externally provided one is used for all cases.

Reply:
Ok, I've created [DAFFODIL-2481](https://issues.apache.org/jira/browse/DAFFODIL-2481) for that issue. I don't think it needs to be fixed as part of the PR, which is only about the new direction stuff.